proto/cosmos/base/reflection/v1beta1/reflection.proto

```diff
@@ -21,6 +25,187 @@ service ReflectionService {
+  string fullname = 1;
+  // field_descriptor_names is a list of the protobuf name (not fullname) of the field
+  // which contains the interface as google.protobuf.Any (the interface is the same, but
+  // it can be in multiple fields of the same proto message)
+  repeated string field_descriptor_names = 2;
+}
+
+// ConfigurationDescriptor contains metadata information on the sdk.Config
+message ConfigurationDescriptor {
+  // bech32_account_address_prefix is the account address prefix
+  string bech32_account_address_prefix = 1;
+  // bech32_validator_address_prefix is the bech32 prefix for validator addresses
+  string bech32_validator_address_prefix = 2;
+  // bech32_consensus_address_prefix is the bech32 prefix for consensus nodes addresses
+  string bech32_consensus_address_prefix = 3;
+  // bech32_account_public_key_prefix is the bech32 prefix for accounts public keys
+  string bech32_account_public_key_prefix = 4;
+  // bech32_validator_public_key_prefix is the bech32 prefix for validators public keys
+  string bech32_validator_public_key_prefix = 5;
+  // bech32_consensus_public_key_prefix is the bech32 prefix for consensus nodes public keys
+  string bech32_consensus_public_key_prefix = 6;
+  // purpose is the BIP-0044 purpose code
+  uint32 purpose = 7;
+  // coin_type is the BIP-0044 coin type code
+  uint32 coin_type = 8;
+  // full_fundraiser_path is the BIP-0044 prefix (deprecated in favor of full_bip44_path)
+  string full_fundraiser_path = 9;
+  // full_bip44_path is the BIP-0044 prefix
+  string full_bip44_path = 10;
+}
+
+// MsgDescriptor describes a cosmos-sdk message that can be delivered with a transaction
+message MsgDescriptor {
+  // msg contains a descriptor of sdk.ServiceMsg or sdk.Msg
+  oneof msg {
+    // service_msg is used when the message is an sdk.ServiceMsg type
+    ServiceMsgDescriptor service_msg = 1;
+    // legacy_msg is used when the message is an sdk.Msg type
+    LegacyMsgDescriptor legacy_msg = 2;
+  }
+}
+
+// ServiceMsgDescriptor describes an sdk.ServiceMsg type
+message ServiceMsgDescriptor {
+  // fullname is the protobuf fullname of the given sdk.ServiceMsg
+  string fullname = 1;
+  // route is the sdk.ServiceMsg route, it is equal to type_url
+  string route = 2;
+  // type_url defines the type URL used when marshalling the type as any
+  // this is required so we can provide type safe google.protobuf.Any marshalling and
+  // unmarshalling, making sure that we don't accept just 'any' type
+  // in our interface fields
+  string type_url = 3;
+}
+
+// LegacyMsgDescriptor describes an sdk.Msg type
+message LegacyMsgDescriptor {
+  // fullname is the protobuf fullname of the message
+  string fullname = 1;
+  // type_url defines the type URL used when marshalling the type as any
+  // this is required so we can provide type safe google.protobuf.Any marshalling and
+  // unmarshalling, making sure that we don't accept just 'any' type
+  // in our interface fields
+  string type_url = 2;
+}
```

Comment:
The plan that was announced with Stargate is that we would be deprecating legacy messages. I would rather not backtrack on this and I'd rather not additional support for things that are going to be deprecated.

Reply:
Ok, but are all supported msgs already service msgs? Because I'd like reflection clients to be able to handle anything that a chain can accept via handlers.

Reply:
All supported Msg's can be represented as service msgs now

Reply:
If all supported msgs are already servicemsgs then this does not need to exist, I thought ATM we were supporting both in a separate way. Thanks for the heads up.